pkg/api/handlers/mcp.go

```diff
@@ -1577,3 +1578,116 @@ func (m *MCPHandler) ClearOAuthCredentials(req api.Context) error {
 	req.WriteHeader(http.StatusNoContent)
 	return nil
 }
+
+func (m *MCPHandler) GetServerDetails(req api.Context) error {
+	_, serverConfig, err := ServerForAction(req)
+	if err != nil {
+		return err
+	}
+
+	details, err := m.mcpSessionManager.GetServerDetails(req.Context(), serverConfig)
+	if err != nil {
+		return err
+	}
+
+	return req.Write(details)
+}
+
+func (m *MCPHandler) StreamServerLogs(req api.Context) error {
+	_, serverConfig, err := ServerForAction(req)
+	if err != nil {
+		return err
+	}
+
+	logs, err := m.mcpSessionManager.StreamServerLogs(req.Context(), serverConfig)
+	if err != nil {
+		return err
+	}
+
+	defer logs.Close()
+
+	// Set up Server-Sent Events headers
+	req.ResponseWriter.Header().Set("Content-Type", "text/event-stream")
+	req.ResponseWriter.Header().Set("Cache-Control", "no-cache")
+	req.ResponseWriter.Header().Set("Connection", "keep-alive")
+	req.ResponseWriter.Header().Set("Access-Control-Allow-Origin", "*")
```

Comment:
Is this header necessary? AI added it, but it seems to me like we shouldn't need to do CORS stuff here.

Reply:
Is it because of the MCP client that are making CORS request from a different domain? I think I also have * set for oauth proxy at some point but I thought it should not be needed.